src/lib/ui.js

```diff
@@ -0,0 +1,126 @@
+import config from '../config'
+import urltemplate from 'url-template'
+import { awaitMessage } from '../utils/async'
+import { getRandCode } from '../utils/rand'
+import { emptyInnerHTML } from '../utils/dom'
+
+const uiMethods = async({ method, href } = {}) => {
+  switch (method) {
+    case 'share':
+      if (typeof href === 'string') {
+        // we didn't do any validation but left it for sharing page to validate it.
+        const dialog = openWindow(encodeURIComponent(href))
+        return awaitMessage(dialog)
+      }
+      throw Error(`href: ${href} is not valid.`)
+
+    default:
+      throw Error(`method: ${method} is not valid`)
+  }
+}
+
+const openWindow = (href = '') => {
+  const SHARE_POST_TO_MIGME_URL = `{+hostUrl}/share_to_migme{?${[
+    'href'
+  ]}}`
+  const data = {
+    href,
+    hostUrl: config.hostUrl,
+  }
+
+  const w = 500
+  const h = 500
+
+  const url = urltemplate.parse(SHARE_POST_TO_MIGME_URL).expand(data)
+
+  const screen = screen || {screen: {left: 0, right: 0}}
+  const dualScreenLeft = window.screenLeft || screen.left
+  const dualScreenTop = window.screenTop || screen.top
+
+  const width = window.innerWidth || document.documentElement.clientWidth || screen.width
+  const height = window.innerHeight || document.documentElement.clientHeight || screen.height
+
+  const left = ((width / 2) - (w / 2)) + dualScreenLeft
+  const top = ((height / 2) - (h / 2)) + dualScreenTop
+
+  return window.open(url, 'migme', `height=${h}, width=${w}, top=${top}, left=${left}, resizable=0`)
+}
+
+const getShareButtonSpanStyle = (layout) => {
+  switch (layout) {
+    case 'button':
+      return 'vertical-align: bottom; width: 58px; height: 20px;'
+    case 'button_count':
+    default:
+      return 'vertical-align: bottom; width: 58px; height: 20px;'
+  }
+}
+
+const getShareButtonIframeStyle = (layout) => {
+  switch (layout) {
+    case 'button':
+      return 'border: none; visibility: visible; width: 58px; height: 20px;'
+    case 'button_count':
+    default:
+      return 'border: none; visibility: visible; width: 58px; height: 20px;'
+  }
+}
+
+export const renderShareButtons = () => {
+  const shareButtonClassName = 'migme-share-button'
+  if (document) {
+    const clientId = config.client_id || ''
+    const containers = document.getElementsByClassName(shareButtonClassName)
```

Comment:
Just inline the class name. Not used in multiple places and makes it easier to read.